### PR TITLE
Fix flipping regressions

### LINKED ISSUE #839: #609 causes regression letting player flip during cutscenes
It's not even been a day and 2.4 already has a regression!

This time, the player is now able to flip during cutscenes thanks to #609 being merged. Specifically, cutscenes that are completely gamestate-based, such as teleporter cutscenes and Level Complete. `git bisect` points to 1eb8570329ec50aac4cb5c772ae71206fd48ff13 being the cause.

### PR DESCRIPTION
**Pull request by leo60228**
## Changes:

Fixes #839 (and https://github.com/TerryCavanagh/VVVVVV/issues/839#issuecomment-911137578).

## Legal Stuff:

By submitting this pull request, I confirm that...

- [x] My changes may be used in a future commercial release of VVVVVV
- [x] I will be credited in a `CONTRIBUTORS` file and the "GitHub Friends"
  section of the credits for all of said releases, but will NOT be compensated
  for these changes

### CODE REVIEW

**Comment by InfoTeddy** [comment]
Will get to this in a bit, already working on fixing #833...

**Review by InfoTeddy — CHANGES_REQUESTED** [review]
This does not fix https://github.com/TerryCavanagh/VVVVVV/issues/839#issuecomment-911137578. I tested it using this level: [mid-air_flip_test.zip](https://github.com/TerryCavanagh/VVVVVV/files/7101334/mid-air_flip_test.zip) Compare it with 2.3, 2.4 current, and your PR.

File: desktop_version/src/Input.cpp [hunk 1]
```diff
@@ -2258,7 +2258,7 @@ void gameinput(void)
             game.gravitycontrol = 1;
             for (size_t ie = 0; ie < obj.entities.size(); ++ie)
             {
-                if (obj.entities[ie].rule == 0)
+                if (obj.entities[ie].rule == 0 && obj.entities[ie].onground)
```
Comment (AllyTally) [inline desktop_version/src/Input.cpp]: You have to check if `onground` or `onroof` is greater than 0.

File: desktop_version/src/Input.cpp [hunk 2]
```diff
@@ -2273,7 +2273,7 @@ void gameinput(void)
             game.gravitycontrol = 0;
             for (size_t ie = 0; ie < obj.entities.size(); ++ie)
             {
-                if (obj.entities[ie].rule == 0)
+                if (obj.entities[ie].rule == 0 && obj.entities[ie].onroof)
```
Comment (AllyTally) [inline desktop_version/src/Input.cpp]: Same here.

**Review by AllyTally — CHANGES_REQUESTED** [review]

**Pull request merged**